### PR TITLE
add support for stop_signal to compose file

### LINKED ISSUE #576: Add an option to the YAML file to specify an alternative signal to stop containers
It would be great if we could specify a signal number in the `fig.yml` file. Fig would then use it with `docker kill -s SIGNAL` instead of `docker stop` when stopping a container.

### PR DESCRIPTION
**Pull request by jstewmon**
Based on #2232
Depends on docker/docker-py#868
Fixes #576

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like something went wrong here, because this PR contains unrelated commits :cry:

**Comment by thaJeztah** [comment]
(I see that https://github.com/docker/docker-py/pull/868 was merged btw)

**Comment by jstewmon** [comment]
@thaJeztah this PR is intentionally based on #2232 (the "unrelated commits") because stop signal support was introduced in version 1.21 of the remote API.

@dnephin  advised basing this PR on #2232 and bumping DEFAULT_API_VERSION to 1.21.

When #2232 is merged, I will rebase on master.

**Comment by thaJeztah** [comment]
@jstewmon ahh, I see now, sorry; usually "unrelated" commits are a result of a bad rebase. Sorry for the noise :smile:

**Comment by jstewmon** [comment]
I've rebased this PR on master, now that v1.21 support has been merged.

File: tests/acceptance/cli_test.py [hunk 1]
```diff
@@ -712,6 +712,19 @@ def test_stop(self):
         self.assertEqual(len(service.containers(stopped=True)), 1)
         self.assertFalse(service.containers(stopped=True)[0].is_running)
 
+    def test_stop_signal(self):
+        self.base_dir = 'tests/fixtures/stop-signal-composefile'
+        self.require_api_version('1.21')
```
Comment (dnephin) [inline tests/acceptance/cli_test.py]: This should be removed now that `1.21` is the default

File: tests/integration/service_test.py [hunk 2]
```diff
@@ -877,6 +877,13 @@ def test_empty_labels(self):
         for name in labels_dict:
             self.assertIn((name, ''), labels)
 
+    def test_stop_signal(self):
+        self.require_api_version('1.21')
```
Comment (dnephin) [inline tests/integration/service_test.py]: same here

**Comment by dnephin** [comment]
I think this looks good, just a couple small changes to the test

**Comment by jstewmon** [comment]
@dnephin , Thanks for the quick review. I've removed those extraneous lines.

**Comment by dnephin** [comment]
This needs another rebase. I believe you'll need to add this to the v2 schema as well.

**Comment by jstewmon** [comment]
Rebased and updated both service schema versions.

**Comment by dnephin** [comment]
There's a test failure after the rebase

**Comment by jstewmon** [comment]
The tests fail because the docker-py version being used in CI is 1.6, and this PR depends on docker/docker-py#868 which is slated for docker-py 1.7.

The tests pass against the docker-py master branch.

I looked through the boilerplate code for tests to see if there was a way of skipping the test based on the version of module, but didn't see one.

Is there a standard way of handling yet-to-be-released dependencies on docker-py?

**Comment by aanand** [comment]
Use a git URL in `requirements.txt`, like this:

    git+https://github.com/docker/docker-py.git@9deffc45a10bf0d03907002a0b3fa6b63c7ff817#egg=docker-py

**Comment by dnephin** [comment]
LGTM!

**Comment by aanand** [comment]
LGTM

**Pull request merged**